### PR TITLE
ID-343-add-date-columns-to-sam-db

### PR DESCRIPTION
**Pull request by gpolumbo-broad**
Ticket: https://broadworkbench.atlassian.net/browse/ID-343

What:

Adds 3 columns to the `sam_user` table:
- createdAt
- registeredAt
- updatedAt

All are GMT datetimestamps.  Tests added to `PostgresDirectoryDAOSpec` to make sure that the dates are correctly being set during user create, register, and updates.  

Why:
Insight into when users were created, registered, and updated should be very useful for troubleshooting and production support.  We also report out metrics on when users register and Sam is the right place to store and report that information from.  

How:
Start with liquibase changes that describe the new columns

---

**PR checklist**

- [ ] I've followed [the instructions](https://github.com/broadinstitute/sam/blob/develop/CONTRIBUTING.md#api-changes) if I've made any changes to the API, _especially_ if they're breaking changes
- [ ] I've filled out the [Security Risk Assessment](https://sdarq.dsp-appsec.broadinstitute.org/jira-ticket-risk-assesment) (requires Broad Internal network access) and attached the result to the JIRA ticket

### CODE REVIEW

**Review by Ghost-in-a-Jar — APPROVED** [review]
this seems pretty reasonable, the only issue I think there might be is that you are adding non-nullable fields without a default? will there be any backfill needed of created_at etc>

File: README.md [hunk 1]
```diff
@@ -78,7 +78,7 @@ Note that in this model Group is a Subject. This allows it to be used interchang
 ### ERD
 ![Sam ERD](sam_erd.png)
 
-The Sam schema has 3 main sections: users and groups in yellow, resources and policies in green, resource type
+The Sam schema can be viewed in sections: users in blue, groups in yellow, resources in orange, policies in green, and resource type
```
Comment (tlangs) [inline README.md:81]: Nice!

**Review by tlangs — APPROVED** [review]

**Comment by gpolumbo-broad** [comment]
jenkins retest this one more time please

**Comment by gpolumbo-broad** [comment]
jenkins retest

**Pull request merged**